core/src/main/scala/akka/persistence/r2dbc/internal/sqlserver/SqlServerQueryDao.scala

```diff
@@ -0,0 +1,207 @@
+@InternalApi
+private[r2dbc] class SqlServerQueryDao(settings: R2dbcSettings, connectionFactory: ConnectionFactory)(implicit
+    ec: ExecutionContext,
+    system: ActorSystem[_])
+    extends PostgresQueryDao(settings, connectionFactory) {
+
+  override def sqlFalse = "0"
+
+  // def because of order of initialization
+  override def log = SqlServerQueryDao.log
+  override protected def sqlDbTimestamp = "SYSUTCDATETIME()"
+
+  override protected val selectEventsSql =
+    sql"""
+      SELECT TOP(@limit) slice, entity_type, persistence_id, seq_nr, db_timestamp, SYSUTCDATETIME() AS read_db_timestamp, event_ser_id, event_ser_manifest, event_payload, writer, adapter_manifest, meta_ser_id, meta_ser_manifest, meta_payload, tags
+      from $journalTable
+      WHERE persistence_id = @persistenceId AND seq_nr >= @from AND seq_nr <= @to
+      AND deleted = $sqlFalse
+      ORDER BY seq_nr"""
+
+  /**
+   * custom binding because the first param in the query is @limit (or '0' when using positional binding)
+   *
+   * Should we use positional binding instead? Named binding is preferred in sqlserver (and slightly cheaper), but this
+   * project uses positional.
+   */
+  override protected def bindSelectEventsSql(
+      stmt: Statement,
+      persistenceId: String,
+      fromSequenceNr: Long,
+      toSequenceNr: Long,
+      bufferSize: Int): Statement =
+    stmt
+      .bind("@limit", settings.querySettings.bufferSize)
+      .bind("@persistenceId", persistenceId)
+      .bind("@from", fromSequenceNr)
+      .bind("@to", toSequenceNr)
+
+  override protected def selectBucketsSql(minSlice: Int, maxSlice: Int): String = {
+    sql"""
+        SELECT TOP(@limit) bucket, count(*) as count from
+         (select DATEDIFF(s,'1970-01-01 00:00:00', db_timestamp)/10 as bucket
+          FROM $journalTable
+          WHERE entity_type = @entityType
+          AND ${sliceCondition(minSlice, maxSlice)}
+          AND db_timestamp >= @fromTimestamp AND db_timestamp <= @toTimestamp
+          AND deleted = $sqlFalse) as sub
+        GROUP BY bucket ORDER BY bucket
+        """
+  }
+
+  override protected def bindSelectBucketsSql(
+      stmt: Statement,
+      entityType: String,
+      fromTimestamp: Instant,
+      toTimestamp: Instant,
+      limit: Int): Statement = {
+    stmt
+      .bind("@limit", limit)
+      .bind("@entityType", entityType)
+      .bindTimestamp("@fromTimestamp", fromTimestamp)
+      .bindTimestamp("@toTimestamp", toTimestamp)
+  }
+
+  override protected def eventsBySlicesRangeSql(
+      toDbTimestampParam: Boolean,
+      behindCurrentTime: FiniteDuration,
+      backtracking: Boolean,
+      minSlice: Int,
+      maxSlice: Int): String = {
+
+    def toDbTimestampParamCondition =
+      if (toDbTimestampParam) "AND db_timestamp <= @until" else ""
+
+    // we know this is a LocalDateTime, so the cast should be ok
+    def localNow: LocalDateTime = timestampCodec.encode(timestampCodec.instantNow()).asInstanceOf[LocalDateTime]
```

Comment:
This smells somewhat. Should we just skip the indirection to the codec here and create it by:
```
LocalDateTime.ofInstant(InstantFactory.now(), UTC)
```